pkg/urbit/vere/king.c

```diff
@@ -1519,8 +1460,11 @@ _king_copy_file(c3_c* src_c, c3_c* dst_c)
     {
       size_t pag_i = 1 << 14;;
       c3_y*  buf_y = c3_malloc(pag_i);
-      ret_i = _king_copy_raw(src_i, dst_i, buf_y, pag_i);
-      err_i = errno;
+      ret_i        = c3_read(src_i, buf_y, pag_i) == pag_i
+                  && c3_write(dst_i, buf_y, pag_i) == pag_i
+                       ? 0
+                       : -1;
+      err_i        = errno;
```

Comment:
I missed this in my earlier review, but this change is just wrong. `pag_i` is just a temporary buffer for the copy, not the length of data to be copied, so we still need the loop. `_king_copy_raw()` should be restored, to something like:

```
static c3_i
_king_copy_raw(c3_i src_i, c3_i dst_i, c3_y* buf_y, size_t pag_i)
{
  ssize_t ret_i;

  do {
    if ( 0 > (ret_i = c3_read(src_i, buf_y, pag_i)) ) {
      return ret_i;
    }

    if ( 0 > (ret_i = c3_write(dst_i, buf_y, (size_t)ret_i)) ) {
      return ret_i;
    }
  }
  while ( ret_i );

  return 0;
}
```

and the callsite here should be something like:

```
if ( 0 > (ret_i = _king_copy_raw(src_i, dst_i, buf_y, pag_i)) ) {
  err_i = -ret_i;
  ret_i = -1;
}
```